addons/account/models/account_move.py

```diff
@@ -2836,7 +2835,7 @@ def button_draft(self):
                 raise UserError(_('You cannot reset to draft an exchange difference journal entry.'))
             if move.tax_cash_basis_rec_id:
                 raise UserError(_('You cannot reset to draft a tax cash basis journal entry.'))
-            if move.restrict_mode_hash_table and move.state == 'posted' and move.id not in excluded_move_ids:
+            if move.restrict_mode_hash_table and move.inalterable_hash and move.id not in excluded_move_ids:
```

Comment:
This `excluded_move_ids` seems like a bad idea: it can potentially break the chain

Reply:
Why do you check `move.restrict_mode_hash_table`?